tests/parser/types/test_dynamic_array.py

```diff
@@ -0,0 +1,430 @@
+from vyper.exceptions import ArrayIndexException, OverflowException
+
+
+def test_list_tester_code(get_contract_with_gas_estimation):
+    list_tester_code = """
+z: DynArray[int128, 3]
+z2: DynArray[DynArray[int128, 2], 2]
+z3: DynArray[int128, 2]
+
+@external
+def foo(x: DynArray[int128, 3]) -> int128:
+    return x[0] + x[1] + x[2]
```

Comment:
What happens when this is called with <3 elements?

Reply:
It clamps against the runtime length of the array (so it would have the same behavior as OOB on a static array or bytestring)